Extension/src/LanguageServer/client.ts

```diff
@@ -3506,12 +3510,40 @@ export class DefaultClient implements Client {
                     character: range.end.character,
                     line: range.end.line
                 }
-            }
+            },
+            copyToClipboard: copy ?? false
         };
 
         const result: CreateDeclarationOrDefinitionResult = await this.languageClient.sendRequest(CreateDeclarationOrDefinitionRequest, params);
-        // TODO: return specific errors info in result.
-        if (result.changes === undefined) {
+        // Create/Copy returned no result.
+        if (result.edit === undefined) {
+            await vscode.window.showInformationMessage(result.errorText); // Copy/Create Declaration/Definition was completely unsuccessful due to api failure.
+            return;
+        }
+
+        // Handle CDD error messaging
+        if (result.errorText) {
+            try {
+                await vscode.env.clipboard.writeText(result.clipboardText);
+                await vscode.window.showInformationMessage(result.errorText + " Declaration/definition was copied to clipboard."); // CDD failed but fallback to clipboard was succesful
```

Comment:
I think this should be changed to an error message with a button added that only does the copy if the user clicks the button.